File: Leet-Code/pow(x,n).py
class Solution:      
    def fastPowers(self, x: float, n: int):
        if(n == 0):
            return 1
        elif n%2 == 0:
            temp = self.fastPowers(x, n/2)
            return (temp*temp)
        else:
            return (x*self.fastPowers(x, n-1))
            

    def myPow(self, x: float, n: int) -> float:
        if(n < 0):
            n = -n
            x = 1/x
        return self.fastPowers(x, n)
        
        
        
        

File: Leet-Code/pow(x,n).cpp
class Solution {
public:
    double fast_power(const double& x, uint n){
        double temp;
        if(n == 0)
            return 1;
        else if (n%2 == 0){
            temp = fast_power(x, n/2);
            return temp*temp;
        }
        else 
            return (x*fast_power(x, n-1));
    }
    
    double myPow(double x, int n) {
        uint n2;
        if(n < 0){
            n2 = -1 * uint(n);
            x = 1/x;
        }
        else
            n2 = n;
        return fast_power(x, n2);
    }
};